app/models/user.js

```diff
@@ -17,12 +17,35 @@ export default DS.Model.extend({
   // relationships
   token: DS.belongsTo('token', {async: true}),
   roles: DS.hasMany('role', {async:true}),
-  organizations: DS.hasMany('organization', {async:true}),
+  //organizations: DS.hasMany('organization', {async:true}),
   sshKeys: DS.hasMany('ssh-key', {async:true}),
 
   // check ability, returns a promise
   // e.g.: user.can('manage', stack).then(function(boolean){ ... });
   can: function(scope, stack){
     return can(this, scope, stack);
-  }
+  },
+
+  organizations: function() {
+    var organizations = [];
+
+    this.get('roles').forEach(function(role) {
+      var organizationId = role.get('_data.links.organization');
+      var organization = organizations.findBy('organizationId', organizationId);
+
+      if(!organization) {
+        organization = {
+          organizationId: organizationId,
+          organization: role.get('organization')
+        };
+        organizations.push(organization);
+      }
+    });
+
+    return organizations.map(function(organization) {
+      return organization.organization;
+    });
+
+  }.property('roles.[]')
```

Comment:
I would simplify:

```javascript
  organizations: function() {
    return this.get('roles').mapBy('organization').uniq();
  }.property('roles.@each.organization')
```

Some references, I know this uses some slick trickery:

* [uniq](http://emberjs.com/api/classes/Ember.Enumerable.html#method_uniq)
* [mapBy](http://emberjs.com/api/classes/Ember.Enumerable.html#method_mapBy)
* [@each](http://emberjs.com/guides/object-model/computed-properties-and-aggregate-data/)

It should have the same effect though, and perform maybe a little better.